msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md

```diff
@@ -274,17 +282,24 @@ app.adaptiveCards.actionSubmit('StaticSubmit', async (context, _state, data: Sub
 
 # [Python](#tab/python4)
 
-[Sample code reference](https://github.com/microsoft/teams-ai/blob/main/python/packages/ai/teams/adaptive_cards/adaptive_cards.py#L129C1-L136C67)
+* [Code sample](https://github.com/microsoft/teams-ai/tree/main/python/samples/03.adaptiveCards.a.typeAheadBot)
+
+* [Sample code reference](https://github.com/microsoft/teams-ai/blob/main/python/samples/03.adaptiveCards.a.typeAheadBot/src/bot.py#L39C1-L78C1)
 
 ```python
-# Use this method as a decorator
-@app.adaptive_cards.action_submit("submit")
-async def execute_submit(context: TurnContext, state: TurnState, data: Any):
-    print(f"Execute with data: {data}")
+@app.message(re.compile(r"static", re.IGNORECASE))
+async def static_card(context: TurnContext, _state: AppTurnState) -> bool:
+    attachment = create_static_search_card()
+    await context.send_activity(Activity(attachments=[attachment]))
     return True
 
-# Pass a function to this method
-app.adaptive_cards.action_submit("submit")(execute_submit)
+@app.adaptive_cards.action_submit("StaticSubmit")
+async def on_static_submit(context: TurnContext, _state: AppTurnState, data) -> None:
+    await context.send_activity(f'Statically selected option is: {data["choiceSelect"]}')
+
+@app.adaptive_cards.action_submit("DynamicSubmit")
+async def on_dynamic_submit(context: TurnContext, _state: AppTurnState, data) -> None:
+    await context.send_activity(f'Dynamically selected option is: {data["choiceSelect"]}')
```

Comment:
Missing extra code here as well from Loop component

Reply:
Updated the complete code snippet.